base/sparse/sparsevector.jl

```diff
@@ -1681,3 +1681,35 @@ droptol!(x::SparseVector, tol, trim::Bool = true) = fkeep!(x, (i, x, tol) -> abs
 
 dropzeros!(x::SparseVector, trim::Bool = true) = fkeep!(x, (i, x, other) -> x != 0, nothing, trim)
 dropzeros(x::SparseVector, trim::Bool = true) = dropzeros!(copy(x), trim)
+
+function _fillnonzero!{Tv,Ti}(arr::SparseMatrixCSC{Tv, Ti}, val)
+    m, n = size(arr)
+    arr.colptr = convert(Vector{Ti}, collect(1:m:n*m+1))
+    arr.rowval = convert(Vector{Ti}, vcat([1:m for _ in 1:n]...))
```

Comment:
SparseMatrixCSC is likely to become an immutable in the future (SparseVector already is) so that would force in place anyway.

Also the array might be pointed to by other variables and this would decouple them.